frontend/src/components/ContentPanel/index.tsx

```diff
@@ -49,25 +49,64 @@ export const ContentPanel: FC<ContentPanelProps> = ({
         setPromptBoxes((prev) => [...prev, newBox]);
     };
 
-    //Callbacks to asynchronously fetch AI data from backend
-    const generateAll = () => {
-        promptBoxes.forEach((p) => {
+    // Generates all the IO boxes that are not locked
```

Comment:
This component is getting pretty gigantic. Maybe some of the stuff there could be moved elsewhere.

Reply:
I'll move the logic to the custom hook 👍